PieceManager/PieceManager.cs

```diff
@@ -1411,20 +1426,25 @@ public static Piece.PieceCategory GetCategory(string name)
             return category;
         }
 
-        var names = Enum.GetNames(typeof(Piece.PieceCategory));
+        if (OtherPieceCategories.TryGetValue(name, out category))
+        {
+            return category;
+        }
+
+        var categories = GetPieceCategoriesMap();
 
-        for (int i = 0; i < names.Length; ++i)
+        foreach (var categoryPair in categories)
         {
-	        if (names[i] == name)
-	        {
-                category = (Piece.PieceCategory)i;
-                PieceCategories[name] = category;
+            if (categoryPair.Value == name)
+            {
+                category = categoryPair.Key;
+                OtherPieceCategories[name] = category;
```

Comment:
A little confused here. Where does OtherPieceCategories get defined or populated? How were you able to compile the changes you made after this to test?

Reply:
forgot to add it to the commit, should compile now. My local code has some further changes to test out the system (cloning the piece multiple times etc) so it got lost in between